### ISSUE TITLE
Broken in coq CI

### ISSUE THREAD

**Issue by SkySkimmer** [UniMath/UniMath#1614]
eg https://gitlab.com/coq/coq/-/jobs/3668196302
Seems to be due to https://github.com/UniMath/UniMath/pull/1610

**Comment by nmvdw** [comment]
Interestingly enough, I don't have this error if I compile the code locally and the error did not occur in the UniMath CI.

Could it be related to the version of Coq?

**Comment by benediktahrens** [comment]
Thanks for filing this issue!

The build fails with
```
COQC UniMath/CategoryTheory/LeftKanExtension.v
File "./UniMath/CategoryTheory/LeftKanExtension.v", line 153, characters 6-29:
Error:
Found no subterm matching "colimIn (lan_colim ?x) ((?M80,, ?M82),, ?M81)
                           · lan_mor ?M79" in the current goal.
Command exited with non-zero status 1
UniMath/CategoryTheory/LeftKanExtension.vo (real: 0.63, user: 0.48, sys: 0.14, mem: 378124 ko)
build/CoqMakefile.make:815: recipe for target 'UniMath/CategoryTheory/LeftKanExtension.vo' failed
make[3]: *** [UniMath/CategoryTheory/LeftKanExtension.vo] Error 1
make[3]: *** Waiting for unfinished jobs....
```
@nmvdw : could you look into this?
@SkySkimmer : which version of Coq does the failed build use? I cannot seem to find the version information.

**Comment by m-lindgren** [comment]
Should we revert #1610 for now, unless we have a solution ready/in the works? So we don't keep coq's CI pipeline broken longer than necessary.

**Comment by benediktahrens** [comment]
Maybe it would be sufficient to remove the offending file on left Kan extensions?

**Comment by m-lindgren** [comment]
Removing `LeftKanExtension.v` hopefully fixes Coq CI. On our side the build will also fail [here](https://github.com/UniMath/UniMath/actions/runs/4026525308/jobs/6921149836#step:5:1981):
```
          coqc UniMath/Bicategories/OtherStructure/Examples/StructureOneTypes.{glob,vo} (exit 1)
  File "./UniMath/Bicategories/OtherStructure/Examples/StructureOneTypes.v", line 604, characters 7-35:
  Error:
  Found no subterm matching "pair_2cell (one_types,, has_binprod_one_types)
                               ?M65 ?M66 (?M67,, ?M68)" in the current goal.
  
          coqc UniMath/Bicategories/OtherStructure/Examples/StructureBicatOfUnivCats.{glob,vo}
          coqc UniMath/Bicategories/OtherStructure/Examples/StructureOneTypes.{glob,vo} (exit 1)
  File "./UniMath/Bicategories/OtherStructure/Examples/StructureOneTypes.v", line 604, characters 7-35:
  Error:
  Found no subterm matching "pair_2cell (one_types,, has_binprod_one_types)
                               ?M65 ?M66 (?M67,, ?M68)" in the current goal.
```

**Comment by nmvdw** [comment]
> Removing `LeftKanExtension.v` hopefully fixes Coq CI. On our side the build will also fail [here](https://github.com/UniMath/UniMath/actions/runs/4026525308/jobs/6921149836#step:5:1981):
> 
> ```
>           coqc UniMath/Bicategories/OtherStructure/Examples/StructureOneTypes.{glob,vo} (exit 1)
>   File "./UniMath/Bicategories/OtherStructure/Examples/StructureOneTypes.v", line 604, characters 7-35:
>   Error:
>   Found no subterm matching "pair_2cell (one_types,, has_binprod_one_types)
>                                ?M65 ?M66 (?M67,, ?M68)" in the current goal.
>   
>           coqc UniMath/Bicategories/OtherStructure/Examples/StructureBicatOfUnivCats.{glob,vo}
>           coqc UniMath/Bicategories/OtherStructure/Examples/StructureOneTypes.{glob,vo} (exit 1)
>   File "./UniMath/Bicategories/OtherStructure/Examples/StructureOneTypes.v", line 604, characters 7-35:
>   Error:
>   Found no subterm matching "pair_2cell (one_types,, has_binprod_one_types)
>                                ?M65 ?M66 (?M67,, ?M68)" in the current goal.
> ```

Interesting. This file hasn't been changed for a while.

**Comment by m-lindgren** [comment]
Seems to be a regression somewhere between `8.16.1` - which compiles `LeftKanExtension.v` fine - and current dev version of Coq. What we can do is merge #1619 and keep our fingers crossed that this fixes Coq CI. 

@SkySkimmer out of curiosity, do you have any idea what changes to Coq may have caused this regression? 

If not it would be interesting to git-bisect Coq between 8.16.1 and current dev against #1610 to see where it breaks.

**Comment by SkySkimmer** [comment]
@JasonGross is there a nice way to minimize with 8.16 vs master?

**Comment by benediktahrens** [comment]
@m-lindgren : it is possible that instead of merging #1619, Coq needs to be fixed.

**Comment by benediktahrens** [comment]
@SkySkimmer : could you advise us on how to proceed? On the one hand, we (UniMath) don't want to keep the Coq CI broken. On the other hand, it seems wrong for us to remove code that works in Coq 8.16 and does not work in Coq master without understanding what the cause is - a bug in 8.16 or an incompleteness in dev.

**Comment by ppedrot** [comment]
I'd recommend fixing the Coq CI as a hotfix and then investigating the issue. It's annoying from the Coq side to get UniMath CI failures, as they can hide additional errors creeping up in the PRs merged into master.

**Comment by nmvdw** [comment]
> I'd recommend fixing the Coq CI as a hotfix and then investigating the issue. It's annoying from the Coq side to get UniMath CI failures, as they can hide additional errors creeping up in the PRs merged into master.

It should be fixed now.

**Comment by JasonGross** [comment]
> @JasonGross is there a nice way to minimize with 8.16 vs master?

Open a draft PR in Coq merging master into v8.16, rebuild the base CI artifacts, then run the auto minimizer?  But if you just do normal minimization with master, it'll probably go fine, even without making sure that 8.16 builds the example fine.

**Comment by SkySkimmer** [comment]
coq/coq@682986916cdc0659594f7d15e94ddb817df793dc is the first bad commit

**Comment by benediktahrens** [comment]
@SkySkimmer : is there an update about this? Or a Coq bug report that we can link to this issue?
Our hope is that once it is clear what caused the regression, we can add back the offending file (possibly in modified form). Thanks a lot!

**Comment by SkySkimmer** [comment]
Should be fixed, you can try reverting the revert

**Comment by m-lindgren** [comment]
@SkySkimmer The problematic file is now merged successfully with #1651. Thank you so much for fixing this!
